consensus/mint-client/src/bin/main.rs

```diff
@@ -10,117 +10,163 @@ use mc_consensus_api::{
 
-            let signer = Ed25519Pair::from(signing_key);
-
-            let tombstone_block = tombstone.unwrap_or_else(|| {
-                let last_block_info = blockchain_api
-                    .get_last_block_info(&Empty::new())
-                    .expect("get last block info");
-                last_block_info.index + MAX_TOMBSTONE_BLOCKS - 1
-            });
-
-            let nonce = nonce.map(|n| n.to_vec()).unwrap_or_else(|| {
-                let mut rng = thread_rng();
-                let mut nonce: Vec<u8> = vec![0u8; NONCE_LENGTH];
-                rng.fill_bytes(&mut nonce);
-                nonce
-            });
-
-            let prefix = MintTxPrefix {
-                token_id,
-                amount,
-                view_public_key: *recipient.view_public_key(),
-                spend_public_key: *recipient.spend_public_key(),
-                nonce,
-                tombstone_block,
+            let tx = params
+                .try_into_mint_tx(|| {
+                    let last_block_info = blockchain_api
+                        .get_last_block_info(&Empty::new())
+                        .expect("get last block info");
+                    last_block_info.index + MAX_TOMBSTONE_BLOCKS - 1
+                })
+                .expect("failed creating tx");
+            let resp = client_api
+                .propose_mint_tx(&(&tx).into())
+                .expect("propose tx");
+            println!("response: {:?}", resp);
+        }
+
+        Commands::GenerateMintTx { out, params } => {
+            let tx = params
+                .try_into_mint_tx(|| panic!("missing tombstone block"))
+                .expect("failed creating tx");
+
+            let json = to_string_pretty(&tx).expect("failed serializing tx");
+
+            fs::write(out, json).expect("failed writing output file");
+        }
+
+        Commands::SubmitMintTx { node, tx_filenames } => {
+            // Load all txs.
+            let txs = tx_filenames
+                .iter()
+                .map(|filename| {
+                    let bytes = fs::read_to_string(filename).unwrap_or_else(|err| {
+                        panic!("Failed reading file {:?}: {}", filename, err)
+                    });
+                    serde_json::from_str(&bytes).unwrap_or_else(|err| {
+                        panic!("Failed parsing tx from file {:?}: {}", filename, err)
+                    })
+                })
+                .collect::<Vec<MintTx>>();
+
+            // All tx prefixes should be the same.
+            if !txs.windows(2).all(|pair| pair[0].prefix == pair[1].prefix) {
+                panic!("All txs must have the same prefix");
+            }
+
+            // Collect all signatures.
+            let mut signatures = txs
+                .iter()
+                .flat_map(|tx| tx.signature.signatures())
+                .cloned()
+                .collect::<Vec<_>>();
+            signatures.sort();
+            signatures.dedup();
```

Comment:
Optional/DRY: Extract this filenames-to-signatures logic to a helper, since it's used in another command